addons/account_asset/models/account_asset_invoice.py

```diff
@@ -79,26 +68,24 @@ def _onchange_asset_category_id(self):
         if self.asset_category_id:
             self.account_id = self.asset_category_id.account_asset_id
 
-    @api.one
     def asset_create(self):
-        if self.asset_category_id and self.asset_category_id.method_number > 1:
-            vals = {
-                'name': self.name,
-                'code': self.invoice_id.number or False,
-                'category_id': self.asset_category_id.id,
-                'value': self.price_subtotal,
-                'partner_id': self.invoice_id.partner_id.id,
-                'company_id': self.invoice_id.company_id.id,
-                'currency_id': self.invoice_id.currency_id.id,
-                'date': self.asset_start_date or self.invoice_id.date_invoice,
-                'invoice_id': self.invoice_id.id,
-            }
-            changed_vals = self.env['account.asset.asset']._onchange_category_id_values(vals['category_id'])
-            vals.update(changed_vals['value'])
-            asset = self.env['account.asset.asset'].create(vals)
-            if self.asset_category_id.open_asset:
-                asset.validate()
-        return True
+        for line in self:
```

Comment:
Actually you change the return type. Maybe someone in the world waits for a "True" result ... pay attention to cleaning, it should not change results and stuff like that.